backend/dataall/api/Objects/Environment/resolvers.py

```diff
@@ -30,16 +31,38 @@ def get_trust_account(context: Context, source, **kwargs):
     return current_account
 
 
-def check_environment(context: Context, source, input=None):
+def get_pivot_role_as_part_of_environment(context: Context, source, **kwargs):
+    ssm_param = ParameterStoreManager.get_parameter_value(region=os.getenv('AWS_REGION', 'eu-west-1'), parameter_path=f"/dataall/{os.getenv('envname', 'local')}/pivotRole/createdAsPartOfEnvironmentStack")
+    return True if ssm_param == "True" else False
+
+
+def check_environment(context: Context, source, account_id, region, pivot_role_as_part_of_environment=False):
+    """ Checks necessary resources for environment deployment.
+    - Check CDKToolkit exists in Account assuming cdk_look_up_role
+    - Check Pivot Role exists in Account if pivot_role_as_part_of_environment is False
+    Args:
+        pivot_role_as_part_of_environment: flag to create pivot role as part of environment stack
+        input: environment creation input
+    """
     ENVNAME = os.environ.get('envname', 'local')
     if ENVNAME == 'pytest':
         return 'CdkRoleName'
-    account = input.get('AwsAccountId')
-    region = input.get('region')
-    cdk_role_name = CloudFormation.check_existing_cdk_toolkit_stack(AwsAccountId=account, region=region)
 
-    if input.get('dashboardsEnabled'):
-        existing_quicksight = Quicksight.check_quicksight_enterprise_subscription(AwsAccountId=account)
+    pivot_role_arn = SessionHelper.get_delegation_role_arn(accountid=account_id)
+    cdk_look_up_role_arn = SessionHelper.get_cdk_look_up_role_arn(
```

Comment:
this is a technical detail how we check the existence of the cdk_toolkit_stack, can be encapsulated/moved to the CloudFormation.check_existing_cdk_toolkit_stack method

Reply:
I am sorry, I think I wasn't clear enough, it is actually minor, I meant we can move lines 52-54 to the CloudFormation.check_existing_cdk_toolkit_stack method then the code here would be like:

```` cdk_role_name = CloudFormation.check_existing_cdk_toolkit_stack(
        AwsAccountId=account_id, region=region
    )`
    
    and implementation of the CloudFormation.check_existing_cdk_toolkit_stack will take care of obtaining the cdk role arn and making use of it as an internal detail how we check the existence of the cdk toolkit stack.

Reply:
I see, yes, let's keep it simple we do not need to specify the role for this method